Major changes

MoodyMan04 opened the pull request:
type hinting, correct raise errors, etc.
changes seen in:
avatar, game_object, item, tile, station, occupiable_station, interact_controller, movement_controller.

occupiable_station now uses super().__init__() from Tile and Station class.

MoodyMan04 committed:
Major changes

type hinting, correct raise errors, etc.
changes seen in:
avatar, game_object, item, tile, station, occupiable_station, interact_controller, movement_controller.

occupiable_station now uses super().__init__() from Tile and Station class.

JeanVArdent reviewed (approved):
lgtm

MoodyMan04 committed:
Update movement_controller.py

changed movement_controller to reflect changes made to occupiable_station and tile, mainly type hinting.

JeanVArdent reviewed (approved):
lgtm

ESatrom reviewed (changes requested):
I was under the impression we wanted to prevent stuff from being set to None?

The setter properties for held_item, score, and position in Avatar seem to allow None despite it not being included in the type hinting. Item looks good for that. Tile has the same issue with occupied_by. Station has the same issue with item.

These should all either be type-hinted as "type|None" or have the setter properties changed to prevent None sets. It is also entirely possible that I'm misunderstanding setter properties and they prevent None sets by default, but I doubt this is the case.

otherwise it looks good

MoodyMan04 committed:
Extra type hinting + fixes

Allowing item, held_item, and position to be None through extra type hinting.

score should only be int: now throws error when score is None.

JeanVArdent commented on game/common/avatar.py:
```diff
@@ -7,59 +7,63 @@
 class Avatar(GameObject):
     def __init__(self, item: Item = None, position: tuple[int,int]=None):
```
Type hint Item|None, tuple[int, int]|None

JeanVArdent commented on game/common/avatar.py:
```diff
@@ -7,59 +7,63 @@
 
     @held_item.setter
-    def held_item(self, item: Item):
+    def held_item(self, item: Item) -> None:
```
Item|None

JeanVArdent commented on game/common/avatar.py:
```diff
@@ -7,59 +7,63 @@
-    def position(self, position: tuple[int, int]):
-        if position and not(isinstance(position, tuple) and list(map(type, position)) == [int, int]):
-            raise ValueError(f"{self.__class__.__name__}.position must be a tuple of two ints.")
+    def position(self, position: tuple[int, int]) -> None:
```
tuple[int, int]|None

JeanVArdent commented on game/common/avatar.py:
```diff
@@ -7,59 +7,63 @@
-        temp: Item = data['held_item']
-        if temp is None:
+        self.score: int = data['score']
+        self.position: tuple[int,int] = data['position']
```
tuple[int,int]|None

JeanVArdent commented on game/common/avatar.py:
```diff
@@ -7,59 +7,63 @@
-        if temp is None:
+        self.score: int = data['score']
+        self.position: tuple[int,int] = data['position']
+        held_item: Item = data['held_item']
```
Item|None

JeanVArdent commented on game/common/map/tile.py:
```diff
@@ -7,36 +7,36 @@
-            self.__occupied_by = occupied_by  
-        else:
-            self.__occupied_by = None 
+    def occupied_by(self, occupied_by: GameObject) -> None:
```
GameObject|None

JeanVArdent commented on game/common/map/tile.py:
```diff
@@ -7,36 +7,36 @@
         super().from_json(data)
-        occupied_by = data['occupied_by']
-        if not occupied_by:
+        occupied_by: GameObject = data['occupied_by']
```
GameObject|None

JeanVArdent commented on game/controllers/interact_controller.py:
```diff
@@ -21,15 +23,17 @@ def handle_actions(self, client, world):
-            client.avatar.held_item = result
+        x += client.avatar.position[0]
+        y += client.avatar.position[1]
+        stat = world.game_map[y][x].occupied_by
```
stat: Station

JeanVArdent commented on game/controllers/interact_controller.py:
```diff
@@ -21,15 +23,17 @@ def handle_actions(self, client, world):
+        stat = world.game_map[y][x].occupied_by
+
+        if stat is not None and isinstance(stat, Station):
+            result = stat.take_action(client.avatar)
```
result: Item|None

JeanVArdent commented on game/controllers/movement_controller.py:
```diff
@@ -7,10 +10,10 @@ class MovementController(Controller):
+        avatar_x: int = client.avatar.position[0]
+        avatar_y: int = client.avatar.position[1]
+        pos_mod: tuple[int,int] = (0,0)
         match client.action.chosen_action:
```
change to client.action in preparation for the action object changes

JeanVArdent commented on game/controllers/movement_controller.py:
```diff
@@ -24,19 +27,19 @@ def handle_actions(self, world, client):
+        temp: Tile = world.game_map[avatar_y][avatar_x]
         while hasattr(temp.occupied_by, 'occupied_by'):
-            temp = temp.occupied_by
+            temp: Tile = temp.occupied_by
```
Not needed here

JeanVArdent replied:
The type hinting that is

JeanVArdent commented on game/common/stations/occupiable_station.py:
```diff
@@ -3,47 +3,26 @@
 
     # take action method
-    def take_action(self, avatar: Avatar = None):
+    def take_action(self, avatar: Avatar) -> Item|None:
```
This function is not needed as it is inherited from station

JeanVArdent commented on game/common/stations/station.py:
```diff
@@ -6,40 +6,42 @@
 # create Station object from GameObject that allows item to be contained in it
 class Station(GameObject):
-    def __init__(self, item: Item = None):
+    def __init__(self, item: Item = None, **kwargs):
```
Item|None

JeanVArdent reviewed (changes requested):
Just a few things that were overlooked. This doesn't reflect your ability, there are just a lot of style changes that are easy to miss. Let's aim to get it pulled into Master by tmrw, EOD

MoodyMan04 committed:
type hinting fixes + other

added extra type hinting

movement_controller change:
  match client.action.chosen_action: is now match client.action:

check __init__ type hinting for avatar and station, as well as movement_controller.

JeanVArdent reviewed (approved):
Lgtm

ESatrom reviewed (changes requested):
I believe score setter in avatar should be "score is None or isinstance(score, int)". If this belief is false, the error message should be changed to reflect it.

the rest of it looks good

Edit: Nope, that score setter thing is just wrong regardless of what it was intended to be "is None and isinstance()" will lead to dead code, is false by definition.

JeanAEckelberg commented:
Good point it should be not None

MoodyMan04 committed:
fixed score.setter

changed and to or in score.setter
